colossalai/zero/sharded_optim/sharded_optim_v2.py

```diff
@@ -161,7 +160,7 @@ def step(self, *args, **kwargs):
                 # Since p.data is fp32 and p.col_attr.sharded_data_tensor is fp16
 
                 # TODO() optimize this line CPU (fp32) -> GPU (fp16)
-                colo_model_data_tensor_move(p, p.col_attr.sharded_data_tensor)
+                p.col_attr.sharded_data_tensor.reset_payload(p.half().to(torch.cuda.current_device()))
```

Comment:
why not use colo_model_data_tensor_move_inline API.

Reply:
DONE